packages/frontend/core/src/modules/doc-info/services/doc-database-backlinks.ts

```diff
@@ -0,0 +1,155 @@
+        if (!rowExists) {
+          return undefined;
+        }
+        return propertyIds
+          .map<DatabaseValueCell>(id => {
+            return {
+              id,
+              value$: signalToLiveData(
+                dataSource.cellValueGet$(rowId, id)
+              ).distinctUntilChanged(equalComparator),
+              property: {
+                id,
+                type$: signalToLiveData(dataSource.propertyTypeGet$(id)),
+                name$: signalToLiveData(dataSource.propertyNameGet$(id)),
+                data$: signalToLiveData(dataSource.propertyDataGet$(id)),
+              },
+            };
+          })
+          .filter((p: any): p is DatabaseValueCell => !!p);
+      })
+    );
+
+    return [hydratedRows$, dataSource] as const;
+  }
+
+  // for each db doc backlink,
+  private watchDatabaseRow$(backlink: {
+    docId: string;
+    rowId: string;
+    databaseBlockId: string;
+    databaseName: string | undefined;
+  }) {
+    return new Observable<DatabaseRow | undefined>(subscriber => {
+      let disposed = false;
+      let unsubscribe = () => {};
+      const docRef = this.docsService.open(backlink.docId);
+      const run = async () => {
+        await this.ensureDocLoaded(backlink.docId);
+        if (disposed) {
+          return;
+        }
+        const maybeDatabaseBlock = docRef.doc.blockSuiteDoc.getBlock(
+          backlink.databaseBlockId
+        );
+        if (maybeDatabaseBlock?.flavour === 'affine:database') {
+          const dbModel = maybeDatabaseBlock.model as DatabaseBlockModel;
+          const [cells$, dataSource] = this.adaptRowCells(
+            dbModel,
+            backlink.rowId
+          );
+          const subscription = cells$.subscribe(cells => {
+            if (cells) {
+              subscriber.next({
+                cells,
+                id: backlink.rowId,
+                doc: docRef.doc,
+                docId: backlink.docId,
+                databaseId: dbModel.id,
+                databaseName: dbModel.title.yText.toString(),
+                dataSource: dataSource,
+              });
+            } else {
+              subscriber.next(undefined);
+            }
+          });
+          unsubscribe = () => subscription.unsubscribe();
+        } else {
+          subscriber.next(undefined);
+        }
+      };
+
+      run().catch(e => {
+        console.error(`failed to get database info:`, e);
+        docRef.release();
+      });
+
+      return () => {
+        docRef.release();
+        disposed = true;
+        unsubscribe();
+      };
+    });
+  }
+
+  // backlinks (docid:blockid:databaseBlockId)
+  //   -> related db rows (DatabaseRow[])
+  watchDbBacklinkRows$(docId: string) {
+    return this.docsSearchService.watchDatabasesTo(docId).pipe(
+      distinctUntilChanged(equalComparator),
+      switchMap(backlinks => {
+        return combineLatest(
+          backlinks.map(backlink => {
+            return this.watchDatabaseRow$(backlink);
```

Comment:
watchDbBacklinkRows will cause open doc to put pressure on storage. Can it be delayed until the user clicks on the collapsed area of ​​the database?

Reply:
yes, it is already implemented like you said. Data loading will only happen when the cell is being rendered (which is collapsed in doc info and collpased in database properties section by default)